### PR TITLE
fix(docker): use 0 timestamp instead of now

### PR DESCRIPTION
**Pull request by janisz**
## Description

This PR brings the "old" behaviour so instead of `now()` we will use `0` timestamp.
https://github.com/stackrox/stackrox/pull/8755/files/2adc41a3e689208ef713d030b14083c679a48031#r1408252161

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/registries/docker/metadata_v1.go [hunk 1]
```diff
@@ -41,10 +43,18 @@ func convertImageToDockerFileLine(img *v1.Image) *storage.ImageLayer {
 	line := strings.Join(img.Config.Cmd, " ")
 	line = strings.Join(strings.Fields(line), " ")
 	instruction, value := lineToInstructionAndValue(line)
+	created := time.Time{}
```
Comment (connorgorman) [inline pkg/registries/docker/metadata_v1.go:46]: var created time.Time

**Review by connorgorman — APPROVED** [review]
small nit 👍

File: pkg/registries/docker/metadata_v1.go [hunk 2]
```diff
@@ -41,10 +43,18 @@ func convertImageToDockerFileLine(img *v1.Image) *storage.ImageLayer {
...
 	line = strings.Join(strings.Fields(line), " ")
 	instruction, value := lineToInstructionAndValue(line)
+	var created time.Time
+	if img.Created != nil {
+		created = *img.Created
+	}
+	protoTS, err := types.TimestampProto(created)
+	if err != nil {
+		log.Error(err)
+	}
```
Comment (RTann) [inline pkg/registries/docker/metadata_v1.go:53]: [`protoconv.ConvertTimeToTimestampOrNil`](https://github.com/stackrox/stackrox/blob/4.3.0/pkg/protoconv/time.go#L50)

**Review by RTann — COMMENTED** [review]
There are still two other instances of `protoconv.ConvertTimeToTimestampOrNow` which I don't think should be there. If we cannot convert the time, we shouldn't just say the layer or image was created right now. We should just say we don't know. We know it was not created right at this moment, and I don't think it makes sense to say something we know is false

File: pkg/registries/docker/metadata_v1.go (hunk 2 above)
Comment (janisz) [reply pkg/registries/docker/metadata_v1.go:53]: https://github.com/stackrox/stackrox/pull/8912

**Pull request merged**